pkg/testutils/logger_test.go

```diff
@@ -45,3 +46,40 @@ func TestJSONLineError(t *testing.T) {
 	_, ok := log.JSONLine(0)["error"]
 	assert.True(t, ok, "must have 'error' key")
 }
+
+var keepLogging = func(logger *zap.Logger, quit chan struct{}) {
+	for {
+		select {
+		case <-quit:
+			return
+		default:
+			logger.Info("string")
+		}
+	}
+}
+
+var keepReading = func(buffer *Buffer, quit chan struct{}) {
+	for {
+		select {
+		case <-quit:
+			return
+		default:
+			buffer.Lines()
+			buffer.Stripped()
+		}
+	}
+}
+
+// This test should complete without any panic
+func TestRaceCondition(t *testing.T) {
+	logger, buffer := NewLogger()
+	logQuit := make(chan struct{})
+	readQuit := make(chan struct{})
+	defer close(logQuit)
+	defer close(readQuit)
+
+	go keepLogging(logger, logQuit)
+	go keepReading(buffer, readQuit)
+
+	time.Sleep(time.Second)
```

Comment:
not sure what you mean by "regression". Do you mean bugs like that? I think it's ok to have a test that makes concurrent access, it just shouldn't run that long. And it would be good to verify if running test with `-race` would've caught the issue (before your fix) with a simpler concurrency test:

```
main:

start := make(chan struct{})
finish := sync.WaitGroup{}
finish.Add(2)

go func() { // reader
    _ = <- start
    // do read
    finish.Done()
}()

go func() {/*writer, similar*/}

close(start)  // unblock goroutines
finish.Wait() // wait for them to finish
```